test/any/web_apis/api_compat_data-test.es6.js

```diff
@@ -56,3 +56,94 @@ describe('AbstractApiCompatData', () => {
         apiName: 'a',
       }));
 });
+
+describe('CompatClassGenerator', () => {
+  let CompatProperty;
+  let Release;
+  let generator;
+  beforeEach(() => {
+    CompatProperty = org.chromium.apis.web.CompatProperty;
+    Release = org.chromium.apis.web.Release;
+    generator = org.chromium.apis.web.CompatClassGenerator.create();
+  });
+
+  it('should register a class', () => {
+    generator.generateClass(
+        generator.generateSpec('org.chromium.apis.web', 'CompatClass', []));
+    expect(foam.lookup('org.chromium.apis.web.CompatClass')).toBeDefined();
+  });
+
+  it('should accept a model', () => {
+    const model = foam.json.parse(
+      generator.generateSpec('org.chromium.apis.web', 'CompatClass', []));
+    expect(foam.core.Model.isInstance(model)).toBe(true);
+    generator.generateClass(model);
+    expect(foam.lookup('org.chromium.apis.web.CompatClass')).toBeDefined();
+  });
+
+  it('should create a property for each release',  () => {
+    const release0 = Release.create({
+      browserName: 'Alpha',
+      browserVersion:'1.2.3',
+      osName: 'Zulu',
+      osVersion: '9.8.7',
+    });
+    const release1 = Release.create({
+      browserName: 'Beta',
+      browserVersion:'3.2.1',
+      osName: 'Yankee',
+      osVersion: '7.8.9',
+    });
+    const Cls = generator.generateClass(
+        generator.generateSpec('org.chromium.apis.web', 'CompatClass', [
+          release0,
+          release1,
+        ]));
+    const props = Cls.getAxiomsByClass(CompatProperty);
+    expect(props.length).toBe(2);
+    expect(foam.util.equals(props[0].release, release0)).toBe(true);
```

Comment:
Add verification of property names

Reply:
My thinking here was that the naming convention is an implementation detail. The thing we care to test is the association with the releases. I.e., suppose some weird use of characters in browser versions causes us to change the naming convention; not sure that should require an update of this test.

I suppose I can see it the other way though: something that requires a change in the naming convention probably surfaces as a bug, and we should lock in the fix with a (modified) test.

Reply:
It depends on how this is going to be used. Unintentionally changing things that other code depends on would be bad, which a test for the name generator would be more likely to catch.

Reply:
I added some more expectations for this.